iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/setup/DirectMethodsCommon.java

```diff
@@ -228,6 +244,98 @@ public void openDeviceClientAndSubscribeToMethods() throws Exception
             null);
     }
 
+    public void SubscribeToMethodAndReceiveAsJsonString() throws Exception
+    {
+        testInstance.setup();
+        this.testInstance.identity.getClient().open(true);
+
+        this.testInstance.identity.getClient().subscribeToMethods(
+                (methodName, methodData, context) ->
+                {
+                    String methodDataAsJsonString = methodData.getPayloadAsJsonString();
+
+                    System.out.println("Device invoked " + methodName);
+                    DirectMethodResponse methodResponseAsJsonString;
+                    int status;
+                    Object result;
+
+                    try
+                    {
+                        switch (methodName)
+                        {
+                            case METHOD_MODIFY:
+                                result = modifyPayload(methodDataAsJsonString);
+                                status = UUID.randomUUID().hashCode();
+                                break;
+                            default:
+                                result = "unknown:" + methodName;
+                                status = METHOD_NOT_DEFINED;
+                                break;
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        result = e.toString();
+                        status = METHOD_THROWS;
+                    }
+
+                    methodResponseAsJsonString = new DirectMethodResponse(status, result);
+
+                    directMethodPayload = methodData;
+                    directMethodResponse = methodResponseAsJsonString;
+                    statusCode = status;
```

Comment:
These tests run in parallel at the gate, so don't assign class level instance variables values that are supposed to be tied to a particular test method. See the DirectMethodTestInstance class in this class for where to save values like these